persistence/src/main/scala/com/mesosphere/usi/storage/zookeeper/AsyncCuratorBuilderFactory.scala

```diff
@@ -0,0 +1,267 @@
+package com.mesosphere.usi.storage.zookeeper
+
+import org.apache.curator.framework.CuratorFramework
+import org.apache.curator.framework.imps.CuratorFrameworkState
+import org.apache.curator.x.async.AsyncCuratorFramework
+import org.apache.curator.x.async.api._
+import org.apache.zookeeper.CreateMode
+import org.apache.zookeeper.data.ACL
+
+import scala.collection.JavaConverters
+
+/**
+  * Simple case class holding factory's configuration options. Sensible defaults are provided for all options.
+  *
+  * @param createOptions a set with create options. By default node parents are created if needed and node data is compressed
+  * @param deleteOptions a set with delete options. By default node children are deleted if the node doesn't exist the delete method will appear to succeed
+  * @param existsOptions a set with exists options. By default no options are set
+  * @param createMode node create mode. By default [[CreateMode.PERSISTENT]] nodes are created
+  * @param acl node ACL. By default an empty list is used
+  * @param nodeVersion node version used to update node data or delete it. By default, -1 is used which matches all versions
+  * @param compressedData true if node data compression is enabled. Compression is enabled by default
+  */
+case class AsyncCuratorBuilderSettings(
```

Comment:
Hm, should we load them from `reference.conf` and `application.conf` instead? @zen-dog already showed a way.

Reply:
I thought about it but this is so low-level - I don't expect the users to change this  **ever**. I'd be in favor of leaving it like this for now and add a `conf` later should that ever be necessary.